armi/operators/operator.py

```diff
@@ -618,6 +631,49 @@ def interactAllCoupled(self, coupledIteration):
         activeInterfaces = [ii for ii in self.interfaces if ii.enabled()]
         self._interactAll("Coupled", activeInterfaces, coupledIteration)
 
+    def _tightCouplingOldValues(self):
+        activeInterfaces = [ii for ii in self.interfaces if ii.enabled()]
+        for interface in activeInterfaces:
+            interface.tightCouplingOldValue = interface.getTightCouplingValue()
+
+    def _computeTightCouplingConvergence(self):
+        activeInterfaces = [ii for ii in self.interfaces if ii.enabled()]
+        convergence = []
+        for interface in activeInterfaces:
+            if interface.tightCouplingOldValue is not None:
```

Comment:
just move the code below 643 into Interface class to compute that interface's convergence. and then let the hook referred to above be responsible for running this.

Reply:
This was also done in https://github.com/terrapower/armi/pull/1033/commits/e01ecd0f21ec6d2911cf6485e633dd09646f8c5e. The convergence is calculated in interfaces.TightCoupler. The loop over active interfaces was split off into a protected method (called from interactAllCoupled) to accommodate unit testing. See the commit message for more details. Hopefully this is closer to what you were thinking 🤞